trunion/ephemeral.py

```diff
@@ -0,0 +1,93 @@
+# ***** BEGIN LICENSE BLOCK *****
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+# ***** END LICENSE BLOCK *****
+
+
+import time
+from M2Crypto import ASN1, EVP, RSA, X509
+
+
+class EphemeralCA(object):
+    """
+    A convenience object that tries to encompass the majority of the functions
+    associated with a certificate authority.
+    """
+    def __init__(self, privkey, certificate, settings, extensions):
+        # Key and certificate are loaded in trunion.crypto.KeyStore's methods
+        # so we expect EVP.Pkey and X509.X509 alike objects
+        self.key = privkey
+        self.certificate = certificate
+        self.settings = settings
+        self.extensions = extensions
+
+    def set_validity_period(self, cert):
+        now = long(time.time())
+        asn1 = ASN1.ASN1_UTCTIME()
+        asn1.set_time(now)
+        cert.set_not_before(asn1)
+        asn1 = ASN1.ASN1_UTCTIME()
+        asn1.set_time(now + self.settings['cert_validity_lifetime'] * 24 * 60 * 60)
+        cert.set_not_after(asn1)
+
+    def certify(self, req):
+        pubkey = req.get_pubkey()
+        cert = X509.X509()
+        cert.set_pubkey(pubkey)
+        cert.set_version(2)  # 2 means X509v3
+        #
+        # We are explicitly using Python's default time type * 1000 to include
+        # milliseconds.  While I don't expect to be generating these more often
+        # than once a second I've be wrong before.
+        #
+        cert.set_serial_number(int(time.time() * 1000))
+        self.set_validity_period(cert)
+
+	cert.set_subject(req.get_subject())
```

Comment:
this code indentation is off which looks like it would raise an IndentationError in Python. Do you have tests covering this?